Resilient bugreport: add timeout to commands + write more stuff

HereThereBeDragons opened the pull request:
Issue #3764 

- subcommands in bugreport now timeout after default 90sec (can be changed by running `cvmfs_config bugreport <timeout_in_sec>`
- add all files listed in `/var/run/cvmfs` (if available will include guard_file used for reload etc)

HereThereBeDragons committed:
cvmfs bugreport: add timeout to subcommands and also add /var/run/cvmfs files

HereThereBeDragons committed:
bugreport: change timeout for each command to 90sec

vvolkl commented:
Thanks!  Did you check how `timeout` behaves if `cmd` does not respond to a SIGKILL? That's a possibility for df for example, when the cvmfs process is in D state.

HereThereBeDragons committed:
let timeout send SIGKILL to get out of stuck processes (works for strace on "D" state processes)

DrDaveD commented:
In my experience I have always had to kill the parent process of the stuck command process, because the stuck command itself becomes unkillable even with -9.  You might need to use something like
```
timeout -s 9 ${timeout_sec} sh -c "$cmd &
                                   wait"
```
With that if $cmd is unkillable it will still continue by just killing the wait command.

HereThereBeDragons committed:
cmvfs_bugreport: move "df" to end of commands in case it hangs and SIGKILL in timeout does not work; unify common commands used on all platforms; fix problems of "eval find ${CVMFS_CACHE_BASE} "

HereThereBeDragons committed:
add daves suggestion to make the timeout more robust for hanging cmds:
"With that if $cmd is unkillable it will still continue by just killing the wait command."

Co-authored-by: DrDaveD <2129743+DrDaveD@users.noreply.github.com>

vvolkl commented on cvmfs/cvmfs_config:
```diff
@@ -1485,35 +1496,38 @@ cvmfs_bugreport() {
+  cvmfs_cmds=('cvmfs2 --version' 'cvmfs_config probe' 'cvmfs_config status' \
+              'cvmfs_config showconfig' 'cvmfs_config chksetup' \
+              'cvmfs_config stat -v' 'cvmfs_talk internal affairs' \
+              "find ${CVMFS_CACHE_BASE} -maxdepth 1 -exec ls -lah {} ;")
```
```suggestion
              "find ${CVMFS_CACHE_BASE} -maxdepth 1 -exec ls -lah \{\} \;")
```

vvolkl commented on cvmfs/cvmfs_config:
```diff
@@ -1466,6 +1473,10 @@ cvmfs_bugreport() {
 
+  echo "Gathering /var/run/cvmfs"
+  mkdir -p var/run/cvmfs
+  cp -r /var/run/cvmfs var/run/cvmfs
```
```
tar: bugreport/var/run/cvmfs/cvmfs/cvmfs.cvmfs-config.cern.ch: socket ignored
```
Let's silence this  - maybe with a `find -type f -exec cp ...`

DrDaveD replied:
`find` with `-exec` can be really slow since it needs to start a program for every file.  I think it would be better to redirect stderr to /dev/null.

vvolkl replied:
Yes, that works too. Although I expect there to be 1-3 files per repo, so performance isn't a big concern

HereThereBeDragons replied:
sadly not a single line replacement to just surpress the warning

HereThereBeDragons committed:
cvmfs_bugreport: bugfix for "find" and surpress copying sockets from /var/run/cvmfs

vvolkl committed:
Merge branch 'devel' into resilient_bugreport_3764
